src/WebJobs.Extensions.DurableTask/Correlation/TelemetryActivator.cs

```diff
@@ -64,23 +64,34 @@ public ValueTask DisposeAsync()
         /// <inheritdoc/>
         public void Dispose()
         {
-            this.TelemetryModule.DisposeAsync().AsTask().GetAwaiter().GetResult();
-            this.WebJobsTelemetryModule.DisposeAsync().AsTask().GetAwaiter().GetResult();
+            if (this.TelemetryModule != null)
+            {
+                this.TelemetryModule.DisposeAsync().AsTask().GetAwaiter().GetResult();
+            }
+
+            if (this.WebJobsTelemetryModule != null)
+            {
+                this.WebJobsTelemetryModule.DisposeAsync().AsTask().GetAwaiter().GetResult();
+            }
         }
 
         /// <summary>
         /// Initialize is initialize the telemetry client.
         /// </summary>
         public void Initialize(ILogger logger)
         {
+            this.endToEndTraceHelper = new EndToEndTraceHelper(logger, this.options.Tracing.TraceReplayEvents);
+            this.endToEndTraceHelper.ExtensionWarningAnnouncement(
```

Comment:
I recall that we discussed only logging this warning if distributed tracing V2 was not explicitly enabled or disabled. However, it looks like we're logging this no matter what for all function apps.